actions/submit-signing-request/task.ts

```diff
@@ -113,31 +115,47 @@ export class Task {
         }
     }
 
+    // if auto-generated GitHub Actions token (secrets.GITHUB_TOKEN) is used for artifact download,
+    // ensure the workflow continues running until the download is complete.
+    // The token is valid only for the workflow's duration
+    private async ensureSignPathDownloadedUnsignedArtifact(signingRequestId: string): Promise<void> {
+        core.info(`Waiting until SignPath downloaded the unsigned artifact...`);
+        const requestData = await (executeWithRetries<SigningRequestDto>(
+            async () => {
+                const signingRequestDto = await (this.getSigningRequest(signingRequestId)
+                    .then(data => {
+                        if(!data.unsignedArtifactLink  && !data.isFinalStatus) {
+                            core.info(`Checking the download status: not yet complete`);
+                            throw new Error('Retry artifact download status check.');
```

Comment:
I see what you're doing here. I learnt (and believe) that using exceptions as a method of managing the control flow rightfully an anti-pattern. 

Here's an idea on how to make that better (in my opinion): I would suggest changing the `executeWithRetries` function to expect the promise to return a discrimnated object, e.g. `{ retry: true } | { data: <RES> }` and have the function perform retries both in the error case as well as in the `{ retry: true }` case.
I would have the respective promise functions return:
* `{ retry: true }` when an expected retry is necessary (e.g. signing request not yet complete; unsigned artifact not yet downloaded from Github)
* `{ data: <RES> }` with the regular response

I just only noticed that this is the case of the other invocation of `executeWithRetries` also. Just a suggestion though, open to discuss whether it is necessary.

Reply:
I agree. Done.

Reply:
Cool, like it much better. You added a `retryReason` to the `ExecuteWithRetriesResult` interface. It is never used. I would suggest removing it. Otherwise, there should be at least a test that covers it in `utils.test.ts`

Reply:
done